File: ruby-elasticemail-examples/rails_app/app/controllers/emails_controller.rb
# frozen_string_literal: true

##
# POST /send
#
# Request body: { to, subject, message }
# Response:     { success: true, transactionId, messageId }
class EmailsController < ApplicationController
  def send_email
    to = params[:to]
    subject = params[:subject]
    message = params[:message]

    unless to && subject && message
      return render json: { error: "Missing required fields: to, subject, message" }, status: :bad_request
    end

    result = ElasticEmail::EmailsApi.new.emails_transactional_post(
      ElasticEmail::EmailTransactionalMessageData.new(
        recipients: ElasticEmail::TransactionalRecipient.new(to: [to]),
        content: ElasticEmail::EmailContent.new(
          from: FROM,
          subject: subject,
          body: [html_part("<p>#{message}</p>")]
        )
      )
    )

    render json: { success: true, transactionId: result.transaction_id, messageId: result.message_id }
  rescue ElasticEmail::ApiError => e
    render_api_error(e)
  end
end

File: dotnet-elasticemail-examples/MvcApp/Controllers/EmailsController.cs
using ElasticEmail.Api;
using ElasticEmail.Client;
using ElasticEmail.Model;
using Microsoft.AspNetCore.Mvc;

namespace MvcApp.Controllers;

[ApiController]
public class EmailsController : ControllerBase
{
    private readonly EmailsApi _emailsApi;

    public EmailsController(EmailsApi emailsApi)
    {
        _emailsApi = emailsApi;
    }

    [HttpPost("/send")]
    public async Task<IActionResult> Send([FromBody] SendRequest body)
    {
        if (string.IsNullOrEmpty(body.To) || string.IsNullOrEmpty(body.Subject) || string.IsNullOrEmpty(body.Message))
        {
            return BadRequest(new { error = "Missing required fields: to, subject, message" });
        }

        try
        {
            var result = await _emailsApi.EmailsTransactionalPostAsync(new EmailTransactionalMessageData(
                recipients: new TransactionalRecipient(to: new List<string> { body.To }),
                content: new EmailContent(
                    from: Ee.From,
                    subject: body.Subject,
                    body: new List<BodyPart> { new BodyPart(BodyContentType.HTML, $"<p>{body.Message}</p>") })));

            return Ok(new { success = true, transactionId = result.TransactionID, messageId = result.MessageID });
        }
        catch (ApiException e)
        {
            return Ee.ApiError(e);
        }
    }
}

public record SendRequest(string? To, string? Subject, string? Message);
